### PR TITLE
Adding logrotate role to rotate sync logs

### PR DESCRIPTION
**Pull request by prem-fissionhq**
subtask: https://dimagi-dev.atlassian.net/browse/SC-1361
Ticket: https://dimagi-dev.atlassian.net/browse/SC-1313

--
Disabled default logrotation done by supervior module by including below variables in supervisor jobs.
```
stdout_logfile_maxbytes=0
stderr_logfile_maxbytes=0
stdout_logfile_backups=0
stderr_logfile_backups=0
```

Adding role to rotate all logs under {{log_dir}} using logrotate module. Please review and let me know your suggestions. 
Tested in my local environment. Once reviewed I will apply on one of the sync server.

### CODE REVIEW

File: roles/commcare_sync/templates/django/celery_default_supervisor_config.j2 [hunk 1]
```diff
@@ -3,3 +3,7 @@ command = {{ virtualenv_dir }}/bin/celery_default_start
 user = {{ app_user }}
 stdout_logfile = {{ log_dir }}/celery.log
 stderr_logfile = {{ log_dir }}/celery.log
+stdout_logfile_maxbytes=0
```
Comment (czue) [inline roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]: can you add a note explaining that we are using logrotate and linking to these docs to show why this is done? http://supervisord.org/logging.html#activity-log-rotation

File: roles/commcare_sync/tasks/django.yml [hunk 2]
```diff
@@ -64,6 +64,10 @@
     - "{{ static_dir }}"
     - "{{ media_dir }}"
 
+- name: set acl on app log directory.
+  shell: setfacl -d -m u::rwx,g::rw,o::rw {{ item }}
+  with_items:
+    - "{{ log_dir }}"
```
Comment (czue) [inline roles/commcare_sync/tasks/django.yml:70]: do we still need this change?

**Review by czue — COMMENTED** [review]
Is submodules considered the "right" way to import a third party role? Like there's not something simpler we can do to just reference it?

(I don't know the answer to this, just checking)

File: roles/commcare_sync/tasks/django.yml (hunk 2 above)
Comment (prem-fissionhq) [reply roles/commcare_sync/tasks/django.yml:70]: No. Sorry, will remove it.. thanks

File: roles/commcare_sync/templates/django/celery_default_supervisor_config.j2 (hunk 1 above)
Comment (prem-fissionhq) [reply roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]: sure, will add Cory.

**Comment by czue** [comment]
@prem-fissionhq just ping me whenver this is ready

File: roles/commcare_sync/templates/django/celery_default_supervisor_config.j2 (hunk 1 above)
Comment (prem-fissionhq) [reply roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]: Done Cory. Please check.

File: roles/commcare_sync/tasks/django.yml (hunk 2 above)
Comment (prem-fissionhq) [reply roles/commcare_sync/tasks/django.yml]: Removed. thanks

File: roles/commcare_sync/templates/django/celery_default_supervisor_config.j2 (hunk 1 above)
Comment (prem-fissionhq) [reply roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]:
```
Is submodules considered the "right" way to import a third party role? Like there's not something simpler we can do to just reference it?
```
yes, we can do that. shall we do it in that way? As it is simple code which does have much changes I have done in this way.

Comment (czue) [reply roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]: Can we do something like this? https://github.com/dimagi/commcare-cloud/commit/6e71572b15bde810eb8c18153970e492f4ce79dd

Submodules just seem awkward if we can avoid them.

**Review by czue — COMMENTED** [review]
Just one final question.

File: roles/commcare_sync/templates/django/celery_default_supervisor_config.j2 (hunk 1 above)
Comment (prem-fissionhq) [reply roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]: sure, I will do it. Thanks Cory.

Comment (prem-fissionhq) [reply roles/commcare_sync/templates/django/celery_default_supervisor_config.j2:6]: > Can we do something like this? [dimagi/commcare-cloud@6e71572](https://github.com/dimagi/commcare-cloud/commit/6e71572b15bde810eb8c18153970e492f4ce79dd)
> 
> Submodules just seem awkward if we can avoid them.

Updated the playbooks as per your suggestion Cory. Please review. Thanks

**Comment by prem-fissionhq** [comment]
@czue Removed the `roles/ansible-logrotate` folder and importing it as the external role using the requirements.yml file. 
We need to run `ansible-galaxy install -r requirements.yml` as the first step to import all roles from an external location. 

Tested in my local environment, please review my changes. thanks

**Review by czue — APPROVED** [review]
looks good to me, thanks!

**Comment by czue** [comment]
We need to run `ansible-galaxy install -r requirements.yml` - sorry can you also add this to [the docs](https://commcare-sync-ansible.readthedocs.io/) wherever it makes sense? (probably [here](https://commcare-sync-ansible.readthedocs.io/en/latest/getting-started.html) for dev and maybe [here](https://commcare-sync-ansible.readthedocs.io/en/latest/deployment.html) for prod)

**Comment by prem-fissionhq** [comment]
Sure, I can do that, thanks Cory.

**Comment by prem-fissionhq** [comment]
@czue updated the docs. Tested in Vagrant. Please review and let me know if you see any issues 
Thanks.

**Review by czue — APPROVED** [review]
looks good, thanks!

**Comment by prem-fissionhq** [comment]
@czue Applied on the demo2 env and it worked as expected. Please review and let me know if you've any questions. Thanks

**Comment by czue** [comment]
@prem-fissionhq looks good to me. are you ready for this to be merged?

**Comment by prem-fissionhq** [comment]
Yes Cory.

**Comment by czue** [comment]
thanks for all the effort on this!

**Pull request merged**